modules/sugar/src/main/ts/ephox/sugar/api/node/ShadowDom.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Document, Element as DomElement,
+  HTMLElement, HTMLElementTagNameMap,
+  Node as DomNode,
+  ShadowRoot
+} from '@ephox/dom-globals';
+import * as Node from './Node';
+import * as Head from './Head';
+import * as Body from './Body';
+import { Option, Type } from '@ephox/katamari';
+import Element from './Element';
+
+export type RootNode = Element<Document | ShadowRoot>;
+
+export const isShadowRoot = (dos: RootNode): dos is Element<ShadowRoot> =>
+  Node.isDocumentFragment(dos);
+
+export const isDocument = (dos: RootNode): dos is Element<Document> =>
```

Comment:
This already exists in `Node` so having it in 2 places could get confusing. Also passing in a `RootNode` here isn't very useful, as a lot of the time if you've got it to something the specific you already know what it is.

So I'd prefer to leave this out and just keep the `isShadowRoot` as that can still be used if needed to find out if a root node is a Document.

Reply:
The version in Sand has the wrong type for my needs here.

Reply:
Sorry, that should have been `NodeType`, not `Node` (the one that's in Sugar).

Reply:
How so, you're just aliasing it?

Reply:
No, you're right. The version in Node is fine.

Reply:
Done.